### PR TITLE
Update types

### PR DESCRIPTION
**Pull request by pkliczewski**
We need to keep CRD version history as well we were missing status.phase and condition.type.

Bug-Url: https://bugzilla.redhat.com/1868229
Signed-off-by: Piotr Kliczewski <piotr.kliczewski@gmail.com>

### CODE REVIEW

**Comment by pkliczewski** [comment]
test failure not related to this PR

**Review by jakub-dzon — APPROVED** [review]

File: pkg/operator/resources/operator/operator.go [hunk 1]
```diff
@@ -1373,6 +1479,186 @@ func CreateResourceMapping() *extv1.CustomResourceDefinition {
...
+																		"namespace": {
+																			Type: "string",
+																		},
+																	},
+																	Required: []string{"name"},
+																},
+																"type": {
+																	Type: "string",
+																},
+																"volumeMode": {
```
Comment (machacekondra) [inline pkg/operator/resources/operator/operator.go]: This should be part only as part of v1beta1 API, please use v1alpha1 schema before this change was merged.

Comment (pkliczewski) [reply pkg/operator/resources/operator/operator.go]: Good point, my copy of `v1alpha1` was from just before your changes. Will fix, Thanks!

**Review by machacekondra — APPROVED** [review]

**Pull request merged**